### PR TITLE
Improve API Docs

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Comment by MadhukaHarith92** [comment]
@kalaiyarasiganeshalingam  The PR heading is not according to the best practices. Please correct them.

A properly formed Git commit subject line or PR heading should always be able to complete the following sentence:

If applied, this commit/PR will your subject line here

Refer [[1]](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md#submit-your-contribution) for more details.

[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md#submit-your-contribution

File: file-ballerina/Module.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+A `Listener` endpoint can be defined using the mandatory parameter `path` and the optional parameter `recursive` as follows:
+
+```ballerina
+listener file:Listener inFolder = new ({
+    path: "<The directory path>",
+    recursive: false
+});
+```
+
+If the listener needs to monitor subdirectories of the given directory, needs to be set `recursive` to true. The default value of this is false.
```
Comment (MadhukaHarith92) [inline file-ballerina/Module.md:23]:
```suggestion
If the listener needs to monitor subdirectories of the given directory, need to set `recursive` to true. The default value of this is false.
```

File: file-ballerina/Module.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+## Overview
+
+This module provides APIs which perform file, file path, and directory operations and a `Directory Listener` which is used to listen a directory in the local file system.
+
+This provide the interface to create, delete, rename the file/directory, retrieve metadata of the given file, and manipulate 
+filename paths in a way compatible according to the target operating system-defined file paths.
+
+The path of the file/directory will be need to define as either forward slashes or backslashes depending on the operating system.
```
Comment (MadhukaHarith92) [inline file-ballerina/Module.md:8]:
```suggestion
The path of the file/directory needs to be defined with either forward slashes or backslashes depending on the operating system.
```

File: README.md [hunk 3]
```diff
@@ -10,11 +10,11 @@ Ballerina File Library
 The File library is one of the standard library packages of the<a target="_blank" href="https://ballerina.io/"> Ballerina
```
Comment (praneesha) [inline README.md:10]:
```suggestion
The `file` library is one of the standard library packages of the<a target="_blank" href="https://ballerina.io/"> Ballerina
```

File: README.md [hunk 4]
```diff
@@ -10,11 +10,11 @@ Ballerina File Library
...
 </a> language.
 
-This package contains functions to perform file system based operations such as create, delete, rename the file/directory, and retrieve metadata of the file.
+This package provides APIs which perform file, file path, and directory operations and a `Directory Listener` which is used to listen a directory in the local file system.
```
Comment (praneesha) [inline README.md:13]:
```suggestion
This package provides APIs, which perform file, file path, and directory operations, and a `Directory Listener`, which is used to listen to a directory in the local file system.
```

File: README.md [hunk 5]
```diff
@@ -10,11 +10,11 @@ Ballerina File Library
...
 
-For more information go to [The File Module](https://ballerina.io/learn/api-docs/ballerina/file/).
+For more information go to [The File Module](https://docs.central.ballerina.io/ballerina/file/latest).
```
Comment (praneesha) [inline README.md:15]:
```suggestion
For more information, go to the [`file` module](https://docs.central.ballerina.io/ballerina/file/latest).
```

File: file-ballerina/Module.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+## Overview
+
+This module provides APIs which perform file, file path, and directory operations and a `Directory Listener` which is used to listen a directory in the local file system.
```
Comment (praneesha) [inline file-ballerina/Module.md:3]:
```suggestion
This module provides APIs, which perform file, file path, and directory operations, and a `Directory Listener`, which is used to listen to a directory in the local file system.
```

Comment (praneesha) [inline file-ballerina/Module.md:3]:
```suggestion
This module provides APIs, which perform file, file path, and directory operations, and a `Directory Listener`, which is used to listen to a directory in the local file system.
```

File: file-ballerina/Module.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+This provide the interface to create, delete, rename the file/directory, retrieve metadata of the given file, and manipulate
```
Comment (praneesha) [inline file-ballerina/Module.md:5]:
```suggestion
This provides the interface to create, delete, rename the file/directory, retrieve metadata of the given file, and manipulate the
```

File: file-ballerina/Module.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
+## Overview
+
+This module provides APIs which perform file, file path, and directory operations and a `Directory Listener` which is used to listen a directory in the local file system.
+
+This provide the interface to create, delete, rename the file/directory, retrieve metadata of the given file, and manipulate 
+filename paths in a way compatible according to the target operating system-defined file paths.
```
Comment (praneesha) [inline file-ballerina/Module.md:6]:
```suggestion
filename paths in a way that is compatible according to the target  file paths defined by the operating system.
```

File: file-ballerina/Module.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+The path of the file/directory needs to be defined with either forward slashes or backslashes depending on the operating system.
```
Comment (praneesha) [inline file-ballerina/Module.md:8]:
```suggestion
The path of the file/directory needs to be defined with either forward slashes or back slashes depending on the operating system.
```

File: file-ballerina/Module.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+### Directory Listener
+
+The `file:Listener` is used to monitor all the files and subdirectories inside the specified directory. 
+
+A `Listener` endpoint can be defined using the mandatory parameter `path` and the optional parameter `recursive` as follows:
```
Comment (praneesha) [inline file-ballerina/Module.md:14]:
```suggestion
A `Listener` endpoint can be defined using the mandatory `path` parameter  and the optional `recursive` parameter  as follows.
```

File: file-ballerina/Module.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+```ballerina
+listener file:Listener inFolder = new ({
+    path: "<The directory path>",
+    recursive: false
+});
+```
+
+If the listener needs to monitor subdirectories of the given directory, need to set `recursive` to true. The default value of this is false.
```
Comment (praneesha) [inline file-ballerina/Module.md:23]:
```suggestion
If the listener needs to monitor subdirectories of the given directory, `recursive` needs to be set to `true`. The default value of this is `false`.
```

File: file-ballerina/Module.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+A `Service` has the defined remote methods with the `file:FileEvent` and can be exposed via a `Listener` endpoint. 
+When there are changes in the listening directory, the `file:FileEvent` will be triggered with the action of the file 
+such as creating, modifying or deleting.
```
Comment (praneesha) [inline file-ballerina/Module.md:27]:
```suggestion
such as creating, modifying, or deleting. 
```

File: file-ballerina/Module.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+});
+```
+
+If the listener needs to monitor subdirectories of the given directory, need to set `recursive` to true. The default value of this is false.
+
+A `Service` has the defined remote methods with the `file:FileEvent` and can be exposed via a `Listener` endpoint. 
+When there are changes in the listening directory, the `file:FileEvent` will be triggered with the action of the file 
+such as creating, modifying or deleting. 
+
+The remote methods supported by the `Service`:
```
Comment (praneesha) [inline file-ballerina/Module.md:29]:
```suggestion
The remote methods supported by the `Service` are as follows.
```

File: file-ballerina/Module.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+**onCreate:** This method is invoked once a new file is created in the listening directory.
+
+**onDelete:** This method is invoked once an existing file is deleted from the listening directory.
+
+**onDelete:** This method is invoked once an existing file is modified in the listening directory.
+
+The following code sample shows how to create a `Service` with `onCreate` remote method and attach it to the above `Listener` endpoint:
```
Comment (praneesha) [inline file-ballerina/Module.md:37]:
```suggestion
The following code sample shows how to create a `Service` with the `onCreate` remote method and attach it to the above `Listener` endpoint:
```

File: file-ballerina/Package.md [hunk 15]
```diff
@@ -1,12 +1,15 @@
 ## Package Overview
 
-This package contains functions to perform file system based operations such as create, delete, rename the
-file/directory, and retrieve metadata of the file.
+The File library is one of the standard library packages of the<a target="_blank" href="https://ballerina.io/"> Ballerina
```
Comment (praneesha) [inline file-ballerina/Package.md:3]:
```suggestion
The `file` library is one of the standard library packages of the<a target="_blank" href="https://ballerina.io/"> Ballerina
```

File: file-ballerina/Package.md [hunk 16]
```diff
@@ -1,12 +1,15 @@
...
+</a> language.
 
-For an example on the usage of the operations, see the [File Example](https://ballerina.io/learn/by-example/file.html).
+This package provides APIs which perform file, file path, and directory operations and a `Directory Listener` which is used to listen a directory in the local file system.
```
Comment (praneesha) [inline file-ballerina/Package.md:6]:
```suggestion
This package provides APIs, which perform file, file path, and directory operations, and a `Directory Listener`, which is used to listen to a directory in the local file system.
```

File: file-ballerina/file.bal [hunk 17]
```diff
@@ -88,11 +88,11 @@ isolated function getRawMetaData(@untainted string path) returns MetaData|Error
 
 # Returns the metadata information of the file specified in the file path.
 # ```ballerina
-# file:MetaData | error result = file:getMetaData("foo/bar.txt");
+# file:MetaData result = check file:getMetaData("foo/bar.txt");
 # ```
 #
 # + path - String value of the file path.
-# + return - The `MetaData` instance with the file metadata or else an `file:Error`
+# + return - The `MetaData` instance with the file metadata, or else an `file:Error`
```
Comment (praneesha) [inline file-ballerina/file.bal:95]:
```suggestion
# + return - The `MetaData` instance with the file metadata or else a `file:Error`
```

File: file-ballerina/file.bal [hunk 18]
```diff
@@ -110,11 +110,11 @@ isolated function readDirRaw(@untainted string path) returns MetaData[]|Error =
 # Reads the directory and returns a list of metadata of files and directories
 # inside the specified directory.
 # ```ballerina
-# file:MetaData[] | error results = file:readDir("foo/bar");
+# file:MetaData[] results = check file:readDir("foo/bar");
 # ```
 #
 # + path - String value of the directory path.
-# + return - The `MetaData` array or else an `file:Error` if there is an error
+# + return - The `MetaData` array, or else an `file:Error` if there is an error
```
Comment (praneesha) [inline file-ballerina/file.bal:117]:
```suggestion
# + return - The `MetaData` array or else a `file:Error` if there is an error
```

File: file-ballerina/file.bal [hunk 19]
```diff
@@ -144,13 +144,13 @@ public isolated function copy(@untainted string sourcePath, @untainted string de
...
 # ```ballerina
-# string|error tmpFile = file:createTemp();
+# string tmpFile = check file:createTemp();
 # ```
 #
 # + suffix - Optional file suffix
 # + prefix - Optional file prefix
 # + dir - The directory path where the temp file should be created. If not specified, temp file will be created in the default temp directory of the OS.
-# + return - Temporary file path or an error if one occured
+# + return - Temporary file path, or else an `file:Error` if there is an error
```
Comment (praneesha) [inline file-ballerina/file.bal:153]:
```suggestion
# + return - Temporary file path or else a `file:Error` if there is an error
```

File: file-ballerina/file.bal [hunk 20]
```diff
@@ -159,13 +159,13 @@ public isolated function createTemp(string? suffix = (), string? prefix = (), st
...
 # ```ballerina
-# string|error tmpDir = file:createTempDir();
+# string tmpDir = check file:createTempDir();
 # ```
 #
 # + suffix - Optional directory suffix
 # + prefix - Optional directory prefix
 # + dir - The directory path where the temp directory should be created. If not specified, temp directory will be created in the default temp directory of the OS.
-# + return - Temporary directory path or an error if one occured
+# + return - Temporary directory path, or else an `file:Error` if there is an error
```
Comment (praneesha) [inline file-ballerina/file.bal:168]:
```suggestion
# + return - Temporary directory path or else a `file:Error` if there is an error
```

File: file-ballerina/file.bal [hunk 21]
```diff
@@ -184,7 +184,7 @@ public isolated function createTempDir(string? suffix = (), string? prefix = (),
 #  `IS_SYMLINK` - Test whether a file path is a symlink,
 #  `READABLE` - Test whether a file path is readable,
 #  `WRITABLE` - Test whether a file path is writable.
-# + return - True/false depending on the option to be tested, or else an error if one occurs
+# + return - True/false depending on the option to be tested, or else an `file:Error` if there is an error
```
Comment (praneesha) [inline file-ballerina/file.bal:187]:
```suggestion
# + return - True/false depending on the option to be tested, or else an `file:Error` if there is an error
```
```suggestion
# + return - True/false depending on the option to be tested or else a `file:Error` if there is an error
```

**Review by praneesha — APPROVED** [review]

**Review by MadhukaHarith92 — APPROVED** [review]
LGTM

**Pull request merged**